### ISSUE TITLE
Split 3 in cross-validation

### ISSUE THREAD

**Issue by seungjinna** [percolator/percolator#359]
Hello, 

I know that Percolator split the data sets into 3 subsets for 3-fold cross validation. 
In Split 3, I am getting the following errors:
## percolator log ###############################################
Found 2689136 PSMs
Separate target and decoy search inputs detected, using target-decoy competition on Percolator scores.
Train/test set contains 2330931 positives and 358205 negatives, size ratio=6.50725 and pi0=1
Selecting Cpos by cross-validation.
Selecting Cneg by cross-validation.
Split 1:        Selected feature 13 as initial direction. Could separate 1449808 training set positives with q<0.01 in that direction.
Split 2:        Selected feature 13 as initial direction. Could separate 1449808 training set positives with q<0.01 in that direction.
Split 3:        Exception caught: Error in the input data: cannot find an initial direction with positive training examples. Consider setting/raising the initial training FDR threshold (--train-fdr-initial).
Terminating.
#############################################################

I raised the train-fdr-initialI, but it is not resolved.
To disperse the decoys equally, I also shuffled the PSM list, but the issue persists.

Do you have any idea about this error?  
Split 3 was generated in a different way from Split 1 and Split 2?

**Comment by MatthewThe** [comment]
That's quite odd. It's also a bit suspicious that it finds exactly the same number of positives in the first 2 splits. Any chance you can share the input file?

What is feature 13, could you plot a histogram of that feature?

**Comment by seungjinna** [comment]
Thanks for the reply.

You can download the pin file (zipped) via this link,
https://drive.google.com/file/d/11fetbtvORFDUY7kLDy_rY3fBFfyVXDjr/view?usp=sharing

Best
Seungjin

2023년 11월 7일 (화) 오전 2:38, MatthewThe ***@***.***>님이 작성:

**Comment by MatthewThe** [comment]
Dear Seungjin,

The problem is that the `ScanNr` column only contains zeros. Therefore, percolator puts them all in the same cross validation bin, meaning that the other two cross validation bins are empty.

-Matthew

**Comment by seungjinna** [comment]
Thanks for your help.
In my file, the values in the SpecId column were distinct. So I did not
think it mattered if ScanNr had zero values.
Should I fill the column with auto-increment values like a line number if
ScanNr is unavailable?

Best,
Seungjin

2023년 11월 10일 (금) 오후 6:40, MatthewThe ***@***.***>님이 작성:

**Comment by MatthewThe** [comment]
Yes, that would work. 

Just note that we also use the scannr for retaining only one PSM per scan, so if you want to use that behavior, one should provide the same scannr to PSMs from the same scan.
